pkg/iobuf/bufferedreaderseeker.go

```diff
@@ -53,32 +54,52 @@ type BufferedReadSeeker struct {
 	sizeKnown bool  // Whether the total size of the reader is known
 }
 
+// isSeekable checks if a reader reliably supports seeking operations.
+// Some types, like os.File when used as a pipe, may implement io.Seeker
+// but do not actually support seeking.
+func isSeekable(r io.Reader) bool {
+	seeker, ok := r.(io.Seeker)
+	if !ok {
+		return false
+	}
+
+	_, err := seeker.Seek(0, io.SeekCurrent)
+	return err == nil
+}
+
 // NewBufferedReaderSeeker creates and initializes a BufferedReadSeeker.
 // It takes an io.Reader and checks if it supports seeking.
 // If the reader supports seeking, it is stored in the seeker field.
 func NewBufferedReaderSeeker(r io.Reader) *BufferedReadSeeker {
 	const defaultThreshold = 1 << 24 // 16MB threshold for switching to file buffering
 
-	seeker, _ := r.(io.Seeker)
+	var (
+		buf      *buffer.Buffer
+		seeker   io.Seeker
+		seekable bool
+	)
 
-	var buf *buffer.Buffer
-	if seeker == nil {
+	if isSeekable(r) {
+		seeker, _ = r.(io.Seeker)
+		seekable = true
```

Comment:
Since it's only used here, I'd suggest changing `isSeekable` to:

```go
func asSeeker(r io.Reader) io.Seeker { /* ... */ }
```

and using it like:

```go
if s := asSeeker(r); s != nil {
    seeker = s
} else {
    buf = defaultBufferPool.Get()
}
```

or maybe

```go
seeker = asSeeker(r)
if seeker == nil {
    buf = defaultBufferPool.Get()
}
```